src/html.ts

```diff
@@ -3,17 +3,32 @@ import { ResolvedVitePWAOptions } from './types'
 
 export function injectServiceWorker(html: string, base: string, options: ResolvedVitePWAOptions) {
   const basePath = base.startsWith('/') ? `/${base}` : base
-  return html.replace(
-    '</head>',
-    `
-<link rel="manifest" href="${join(basePath, 'manifest.webmanifest')}">
-  <script>
-    if('serviceWorker' in navigator) {
-      window.addEventListener('load', () => {
-        navigator.serviceWorker.register('${join(basePath, options.filename)}', { scope: './' })
-      })
-    }
-  </script>
-</head>`,
-  )
+  const { inline } = options;
+
+  if (inline) {
+    return html.replace(
+      '</head>',
+      `
+  <link rel="manifest" href="${join(basePath, 'manifest.webmanifest')}">
+    <script>
+      if('serviceWorker' in navigator) {
+        window.addEventListener('load', () => {
+          navigator.serviceWorker.register('${join(
+            basePath,
+            options.filename
+          )}', { scope: './' })
+        })
+      }
+    </script>
+  </head>`
+    );
+  } else {
+    return html.replace(
+      '</head>',
+      `
+  <link rel="manifest" href="${join(basePath, 'manifest.webmanifest')}">
+  <script src="registerServiceWorker.js"></script>
```

Comment:
```suggestion
  <script src="registerServiceWorker.js"></script>
```

We shall respect the `base` option

Reply:
done